lib/app/utils/terminology.rb

```diff
@@ -135,7 +138,30 @@ def self.load_validators(directory = 'resources/terminology/validators/bloom')
     end
 
     def self.get_valueset(url)
-      @known_valuesets[url].valueset || raise(UnknownValueSetException, url)
+      @known_valuesets[url] || raise(UnknownValueSetException, url)
+    end
+
+    def self.get_valueset_by_id(id)
+      unless @valueset_ids
+        @valueset_ids = {}
+        @known_valuesets.each_pair do |k, v|
+          @valueset_ids[v&.valueset_model&.id] = k
```

Comment:
What do `k` and `v` actually represent? This makes `@valueset_ids` look like a hash whose keys are `ids`, but what are its values?